corehq/ex-submodules/casexml/apps/phone/utils.py

```diff
@@ -33,54 +33,65 @@
     :param fixture_name: Name of the fixture
-    :param data_fn: Function to generate the XML fixture elements
     :param overwrite_cache: a boolean property from RestoreState object, default is False
     :return: list containing byte string representation of the fixture
     """
-
-    data = None
-    key = '{}/{}'.format(cache_bucket_prefix, domain)
-
-    if not overwrite_cache:
-        data = get_cached_fixture_items(domain, cache_bucket_prefix)
-        _record_datadog_metric('cache_miss' if data is None else 'cache_hit', key)
-
-    if data is None:
-        with CriticalSection([key]):
-            if not overwrite_cache:
-                # re-check cache to avoid re-computing it
-                data = get_cached_fixture_items(domain, cache_bucket_prefix)
-                if data is not None:
-                    return [data]
-
-            _record_datadog_metric('generate', key)
-            items = data_fn()
-            io_data = write_fixture_items_to_io(items)
-            data = io_data.read()
-            io_data.seek(0)
-            cache_fixture_items_data(io_data, domain, fixture_name, cache_bucket_prefix)
+    io_streams = []
+    for cache_bucket_prefix, data_fn in cache_bucket_prefix_data_fn_pairings:
+        io_stream = None
+        key = '{}/{}'.format(cache_bucket_prefix, domain)
+
+        if not overwrite_cache:
+            io_stream = get_cached_fixture_items(domain, cache_bucket_prefix)
+            _record_datadog_metric('cache_miss' if io_stream is None else 'cache_hit', key)
+            if io_stream is not None:
+                io_stream = BytesIO(io_stream)
+
+        if io_stream is None:
+            with CriticalSection([key]):
+                if not overwrite_cache:
+                    # re-check cache to avoid re-computing it
+                    io_stream = get_cached_fixture_items(domain, cache_bucket_prefix)
+                    if io_stream is not None:
+                        io_stream = BytesIO(io_stream)
+                if io_stream is None:
+                    _record_datadog_metric('generate', key)
+                    item = data_fn()
+                    io_stream = BytesIO()
+                    io_stream.write(ElementTree.tostring(item, encoding='utf-8'))
+                    io_stream.seek(0)
+                    cache_fixture_items_data(io_stream, domain, fixture_name, cache_bucket_prefix)
+        io_streams.append(io_stream)
+
+    combined_io_data = combine_fixture_io_streams(io_streams)
+    combined_io_data.seek(0)
+    data = combined_io_data.read()
 
     global_id = GLOBAL_USER_ID.encode('utf-8')
     b_user_id = user_id.encode('utf-8')
     return [data.replace(global_id, b_user_id)] if data else []
 
 
-def write_fixture_items_to_io(items):
+def combine_fixture_io_streams(io_streams):
     io = BytesIO()
     io.write(ITEMS_COMMENT_PREFIX)
-    io.write(six.text_type(len(items)).encode('utf-8'))
+    io.write(six.text_type(len(io_streams)).encode('utf-8'))
```

Comment:
`six` can be removed. It's long obsolete.
```suggestion
    io.write(str(len(io_streams)).encode('utf-8'))
```

FYI, I did not review other changes in this file carefully.

Reply:
[14a224f](https://github.com/dimagi/commcare-hq/pull/36091/commits/14a224fba97017994d42ba9ea38c78b6503b6330)